### PR TITLE
yamlark

### PR DESCRIPTION
**Pull request by kormide**
DO NOT MERGE, still testing

Can't test if this works yet because main seems to be broken.

The parser is kind of disgusting, but it works 🤷‍♂️ .

### CODE REVIEW

File: js/private/test/repositories_checked.bzl [hunk 1]
```diff
@@ -182,15 +182,6 @@ def npm_repositories():
...
             "fs.realpath": ["1.0.0"],
-            "inflight": ["1.0.6"],
-            "inherits": ["2.0.4"],
-            "minimatch": ["3.1.2"],
-            "once": ["1.4.0"],
-            "path-is-absolute": ["1.0.1"],
-            "wrappy": ["1.0.2"],
-            "brace-expansion": ["1.1.11"],
-            "balanced-match": ["1.0.2"],
-            "concat-map": ["0.0.1"],
```
Comment (kormide) [inline js/private/test/repositories_checked.bzl:193]: It's not clear to me why using my yaml parser resulted in some changes to this file.

**Comment by kormide** [comment]
It looks like I need to support sequences. Found a case.

**Comment by kormide** [comment]
Almost there, but there's still a case I'm missing.

File: js/private/pnpm_yaml.bzl [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    elif _is_float(value):
+        return float(value)
+    elif _is_bool(value):
+        return _to_bool(value)
+    else:
+        return value.strip("'")
+
+def _get_current_object(result, stack):
+    object = result
+    for i in range(len(stack)):
```
Comment (alexeagle) [inline js/private/pnpm_yaml.bzl:197]: nit: there's also enumerate, which gives back a list of tuples of (item, index)

**Comment by kormide** [comment]
Oh no.... I did something stupid and I lost my work on supporting sequences 🙀 .

**Pull request closed without merging**